test/test_type_promotion.py

```diff
@@ -430,6 +431,98 @@ def test_transpose(self, device):
         a = torch.tensor([[True, True], [False, True]], device=device)
         self.assertEqual(a.t() == 0, a.t() == False)  # noqa: E712
 
+    def _test_sparse_op_input_tensors(self, device, dtype, coalesced, zeros=True):
```

Comment:
The problem with this function is that, eventually,  we are performing binary ops on tensors that are not-very-sparse. It might be good to generate tensors that are somewhat sparse (50%? 75%?) so that we can comprehensively say that our sparse ops are correct.

Reply:
I agree it'd be preferable to have more sparsity. I could probably use a random mask.